### PR TITLE
Spawn value

### PR DESCRIPTION
**Pull request by ineiti**
personhood/spawner-contract:
- Adds the possibility to spawn a value instance
- Checks that a spawned darc doesn't have other rules than `_sign` and `invoke:darc.evolve`

external/js/cothority:
- adds the value instance
- adds the value instance to the spawner instance

---

🙅‍ Friendly checklist:

- [x] 0. Code comments are added (or updated) when/where needed and explain the WHY of the code.
- [x] 1. Design choices, user documentation and any additional doc are added (or updated) in READMEs.
- [x] 2. Any new behavior is tested and small units of code that can be are unit tested.
- [x] 3. Code comments are added on tests to explain what they do.
- [x] 4. Errors are systematically wrapped, following the `[failed to | counldn't do] ACTION THAT FAILED: " + err.Error()` format.
- [x] 5. Hard limit of 80 chars is always respected.
- [x] 6. Changes are backward compatible.
- [x] 7. Indentation level does not exceed 5, although 4 is already suspicious.
- [x] 8. Functions, files, and packages are kept to a manageable size and decomposed into smaller units if needed.
- [x] 9. There are no magic values.

### CODE REVIEW

File: personhood/contract_spawner.go [hunk 1]
```diff
@@ -80,25 +82,46 @@ func (c *ContractSpawner) Spawn(rst byzcoin.ReadOnlyStateTrie, inst byzcoin.Inst
...
 		if err = c.getCoins(cout, c.CostDarc); err != nil {
 			return
 		}
 		instBuf = inst.Spawn.Args.Search("darc")
 		d, err := darc.NewFromProtobuf(instBuf)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, errors.New("couldn't decode darc: " + err.Error())
+		}
+		// Whitelist allowed darc rules - there is no spawn allowed, and all
```
Comment (jeffallen) [inline personhood/contract_spawner.go]: I really don't understand this. Why is this needed now? Was it wrong before? Is this something about the value contract?

Comment (ineiti) [reply personhood/contract_spawner.go]: The spawner-contract allows any user with coins to create new instances. In order not to have a user create a darc with rules like `spawn:darc` or `invoke:coin.mint`, the allowed rules are restricted.

Chosing between a blacklist and whitelist approach I chose the latter, to avoid that somebody adds an `invoke` to any contract and then opens a security hole in the spawner-contract.

If you prefer, I can create a separate PR for that 'fix' of the spawner-contract.
